sdlib/dmd/gc.d

```diff
@@ -0,0 +1,105 @@
+module dmd.gc;
+
+import d.gc.tcache;
+import d.gc.spec;
+import d.gc.slab;
+import d.gc.emap;
+
+extern(C):
+
+// Some of the bits from druntime.
+enum BlkAttr : uint {
+	FINALIZE = 0b0000_0001,
+	NO_SCAN = 0b0000_0010,
+	APPENDABLE = 0b0000_1000
+}
+
+// ptr: the pointer to query
+// base: => the true base address of the block
+// size: => the full usable size of the block
+// pd: => the page descriptor (used internally later)
+// flags: => what the flags of the block are
+//
+bool __sd_gc_fetch_alloc_info(void* ptr, void** base, size_t* size,
+                              PageDescriptor* pd, BlkAttr* flags) {
+	*pd = threadCache.maybeGetPageDescriptor(ptr);
+	auto e = pd.extent;
+	*flags = cast(BlkAttr) 0;
+	if (e) {
```

Comment:
Try to avoid

```d
if (conditoon) {
  // Very large block...
}

return result;
```

Just flip the condition and return the result right away. This avoids nesting thing deeper and deeper.